---
layout: page
title: research
permalink: /research/
description: at the Wellcome Centre for Human Genetics with Gerton Lunter and Mark McCarthy 
---

Currently I am working at the intersection of deep learning and human genetics to understand how the microbiome may regulate gene expression in Type 2 Diabetes. As a part of this project, we design convolutional neural networks to predict expression and incorporate metagenomic sequence reads to understand independent regulation from the microbiome. 

<div class="img_row">
    <img class="col two left" src="../assets/img/afr.png"/>
    <img class="col one left" src="../assets/img/hmm.png"/>
</div>
<div class="col three caption">
	Inference of alleles which may be derived from Neanderthal introgression.     
</div>

Additionally, I am collaborating to characterize ancient migration events with Monte Carlo methods including [SMCSMC](https://www.biorxiv.org/content/early/2018/08/01/382218). We use traditional approaches for inferring ancestry including $$D$$ statistics and $$F_4$$ ratios alongside actual simulation based inference of ancestral recombination graphs to derive estimates for assymetric migration. 

--- 

### Previous Research

#### @ London School of Hygiene and Medicine ([Frank Dudbridge](https://www2.le.ac.uk/departments/health-sciences/people/staff-pages/professorial-staff/professor-frank-dudbridge))

Building on the idea that a single polygenic risk score may not capture all relevant information, I continued the work of my undergraduate thesis and developed a cardiometabolic risk score for the prediction of various complex traits. I also worked on the theoretical properties of optimal disovery procedure (ODP), a relatively recent framework for multiple testing correction. 

#### @ University of Toronto, Lancaster University ([Jo Knight](http://chicas.lancaster-university.uk/people/knight.html<Paste>))

I wrote [coRge](https://github.com/Chris1221/coRge) to evaluate different methods of adjusting P values for multiple testing correction in GWAS, and also wrote [goldi](https://github.com/Chris1221/goldi) to find Gene Ontology terms in biomedical articles. 

I also collaborated with Nuwan Hettige to construct a polygenic risk score for anti-psychotic dosage and with Meghan Chenoweth to perform a GWAS for NMR, a biomarker for nicotine metabolism. 

#### @ University of Ottawa Heart Institute ([Ruth McPherson](https://www.ottawaheart.ca/physician-researcher-profile/mcpherson-ruth))

I worked alongside Dr. Majid Nikpay to construct polygenic risk scores for dyslipidedmia and found them to be significantly altered by adiposity. Additionally used a Mendelian randomization study to find links between Coronary Artery Disease aetiology and obesity. 

My thesis focused on combining polygenic risk scores across different predisposing conditions to create a master score which would be more predictive than its constituents. This was formalized when in London with Frank Dudbridge. 

#### @ University of Ottawa ([Stephane Aris-Brosou](https://science.uottawa.ca/biology/people/aris-brosou-stephane))

I designed and implemented a rejection sampler to determine phylogeny topology in viral evolution. And started learning `R`!  
